pkg/inspection/consistency.go

```diff
@@ -116,42 +118,69 @@ func (c *Server) CollectMemberConsistency(inspection *kstonev1alpha1.EtcdInspect
 		klog.Errorf("failed to load tls config, namespace is %s, name is %s, err is %v", namespace, name, err)
 		return err
 	}
-
 	path := DefaultInspectionPath
-	annotations := inspection.ObjectMeta.Annotations
-	if annotations != nil {
-		if info, found := annotations[inspectionConsistencyAnno]; found {
-			consistencyInfo := &ConsistencyInfo{}
-			err = json.Unmarshal([]byte(info), consistencyInfo)
-			if err != nil {
-				klog.Errorf("failed to load inspection info, err is %v", err)
-			} else {
-				path = consistencyInfo.Path
-			}
-		}
-	}
-
 	var msg string
-	var nodeKeyDiff uint64
-	ch, errch := c.getEtcdNodeKeyDiff(cluster, path, tlsConfig)
-	err = <-errch
+	nodeConsistencyDiff := make(map[string]uint64)
+	endpointRaftIndexDiff := make(map[string]uint64)
+	endpointConsistencyMap, err := c.getEtcdConsistencyDiff(cluster, path, tlsConfig)
 	if err != nil {
-		msg = fmt.Sprintf("failed to collectEtcdNodeKeyDiff,etcd cluster %s,err is %v", cluster.Name, err)
+		msg = fmt.Sprintf("failed to collectEtcdConsistencyDiff, etcd cluster %s, err is %v", cluster.Name, err)
 		klog.Errorf("%s", msg)
 	} else {
-		var keys []uint64
-		for v := range ch {
-			if v == math.MaxUint64 {
-				break
+		endpointRaftIndex := make(map[string]uint64)
+		endpointRaftAppliedIndex := make(map[string]uint64)
+		nodeConsistencyTotal := make(map[string][]uint64)
+		for endpointConsistency, num := range endpointConsistencyMap {
+			endpointConsistency := strings.Split(endpointConsistency, "_")
+			if len(endpointConsistency) != 2 {
```

Comment:
If  `endpointConsistency`  can ensure both endpoint and consistencyType are not empty string in method `getEtcdConsistencyDiff()`, this length check won't be necessary.

Reply:
deleted this check